src/zenml/models/v2/core/run_template.py

```diff
@@ -0,0 +1,385 @@
+    )
+
+    deployment_id: UUID = Field(
+        title="The deployment that should be the base of the created template."
+    )
+    tags: Optional[List[str]] = Field(
+        default=None,
+        title="Tags of the run template.",
+    )
+
+
+# ------------------ Update Model ------------------
+
+
+class RunTemplateUpdate(BaseUpdate):
+    """Run template update model."""
+
+    name: Optional[str] = Field(
+        default=None,
+        title="The name of the run template.",
+        max_length=STR_FIELD_MAX_LENGTH,
+    )
+    description: Optional[str] = Field(
+        default=None,
+        title="The description of the run template.",
+        max_length=TEXT_FIELD_MAX_LENGTH,
+    )
+    add_tags: Optional[List[str]] = Field(
+        default=None, title="New tags to add to the run template."
+    )
+    remove_tags: Optional[List[str]] = Field(
+        default=None, title="Tags to remove from the run template."
+    )
+
+
+# ------------------ Response Model ------------------
+
+
+class RunTemplateResponseBody(WorkspaceScopedResponseBody):
+    """Response body for run templates."""
+
+    runnable: bool = Field(
+        title="If a run can be started from the template.",
+    )
+    latest_run_id: Optional[UUID] = Field(
+        default=None,
+        title="The ID of the latest run of the run template.",
+    )
+    latest_run_status: Optional[ExecutionStatus] = Field(
+        default=None,
+        title="The status of the latest run of the run template.",
+    )
+    tags: List[TagResponse] = Field(
+        title="Tags associated with the run template.",
+    )
+
+
+class RunTemplateResponseMetadata(WorkspaceScopedResponseMetadata):
+    """Response metadata for run templates."""
+
+    description: Optional[str] = Field(
+        default=None,
+        title="The description of the run template.",
+    )
+    pipeline_version_hash: Optional[str] = Field(
+        default=None, title="The pipeline version hash of the template."
+    )
+    pipeline_spec: Optional[PipelineSpec] = Field(
+        default=None, title="The pipeline spec of the template."
+    )
+    config_template: Optional[Dict[str, Any]] = Field(
+        default=None, title="Run configuration template."
+    )
+
+
+class RunTemplateResponseResources(WorkspaceScopedResponseResources):
+    """All resource models associated with the run template."""
+
+    source_deployment: Optional[PipelineDeploymentResponse] = Field(
+        default=None,
+        title="The deployment that is the source of the template.",
+    )
+    pipeline: Optional[PipelineResponse] = Field(
+        default=None, title="The pipeline associated with the template."
+    )
+    build: Optional[PipelineBuildResponse] = Field(
+        default=None,
+        title="The pipeline build associated with the template.",
+    )
+    code_reference: Optional[CodeReferenceResponse] = Field(
+        default=None,
+        title="The code reference associated with the template.",
+    )
```

Comment:
You mean in the resources of the deployment?

I added them here explicitly to make work in the frontend easier by avoiding extra requests.

Reply:
Yes, I was talking about the resources of the deployment.

Even though I think it's good practice to change it whenever we have the chance, since the resources of a response won't have their own resources, it wouldn't help in this case. We can correct it later on.